Source/ARTWebSocketTransport.m

```diff
@@ -135,27 +137,27 @@ - (void)connectWithToken:(NSString *)token {
     _state = ARTRealtimeTransportStateOpening;
     [self.logger debug:__FILE__ line:__LINE__ message:@"R:%p WS:%p websocket connect with token", _delegate, self];
     NSURLQueryItem *accessTokenParam = [NSURLQueryItem queryItemWithName:@"accessToken" value:token];
-    [self setupWebSocket:@[accessTokenParam] withOptions:self.options resumeKey:self.resumeKey connectionSerial:self.connectionSerial];
+    [self setupWebSocket:@{accessTokenParam.name: accessTokenParam} withOptions:self.options resumeKey:self.resumeKey connectionSerial:self.connectionSerial];
     // Connect
     [self.websocket open];
 }
 
-- (NSURL *)setupWebSocket:(__GENERIC(NSArray, NSURLQueryItem *) *)params withOptions:(ARTClientOptions *)options resumeKey:(NSString *)resumeKey connectionSerial:(NSNumber *)connectionSerial {
-    NSArray *queryItems = params;
-
+- (NSURL *)setupWebSocket:(NSDictionary<NSString *, NSURLQueryItem *> *)params withOptions:(ARTClientOptions *)options resumeKey:(NSString *)resumeKey connectionSerial:(NSNumber *)connectionSerial {
```

Comment:
Why do we accept a dictionary? We only use it one place? We could just accept a `NSURLQueryItem`?

Reply:
No, we need a collection of `NSURLQueryItem` - before it was `NSArray`.

Reply:
I know that it was `NSArray` before, but was it necessary? The function is only being used to accept 1 `NSURLQueryItem`. We are making the code more verbose by requiring users to create an NSDictionary before passing it in.

We only create dictionaries with 1 item, we can just pass the `NSURLQueryItem` directly?

Reply:
Okay, I guess we can keep the parameter to be a dictionary in case callers pass more than 1 param option.